src/main/java/com/github/kaktushose/jda/commands/dispatching/reply/Components.java

```diff
@@ -0,0 +1,66 @@
+package com.github.kaktushose.jda.commands.dispatching.reply;
+
+import com.github.kaktushose.jda.commands.annotations.interactions.Button;
+import com.github.kaktushose.jda.commands.annotations.interactions.EntitySelectMenu;
+import com.github.kaktushose.jda.commands.annotations.interactions.StringSelectMenu;
+import com.github.kaktushose.jda.commands.dispatching.internal.Runtime;
+
+import java.util.Arrays;
+
+/// Represents a component, namely [Button], [StringSelectMenu] or [EntitySelectMenu], that should be added to a reply.
+///
+/// Also holds the following two settings:
+/// - enabled:
+///
+/// to enable or disable the component
+/// - independent:
+///
+/// whether the component should be executed in the same [Runtime] as the command it is bound to or not. If `true`,
+/// every component interaction will create a new [Runtime]. Furthermore, the component cannot expire and will always
+/// get executed, even after a bot restart.
+///
+/// ### Example:
+/// ```
+/// @SlashCommand("example command")
+/// public void onCommand(CommandEvent event) {
+///     event.with().components(Components.of(true, false, "onButton")).reply();
+/// }
+///```
+/// @since 2.3.0
+public record Components(boolean enabled, boolean independent, String name) {
```

Comment:
We should name this `Component` since it's only one component.

Reply:
approved